src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+import com.umbr3114.auth.SessionManager;
+import com.umbr3114.auth.SparkSessionManager;
+import com.umbr3114.auth.UserModel;
+import com.umbr3114.common.RequestParamHelper;
+import com.umbr3114.data.CollectionFactory;
+import com.umbr3114.models.CommentListingModel;
+import com.umbr3114.models.CommentModel;
+import com.umbr3114.models.PostModel;
+import org.mongojack.JacksonMongoCollection;
+import org.bson.types.ObjectId;
+import spark.Route;
+import com.mongodb.client.model.Filters;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+import java.text.SimpleDateFormat;
+
+
+public class CommentController {
+
+    public static Route saveComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        CommentModel commentModel;
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String bodyText = paramHelper.valueOf("bodyText");
+        String authorId = sessionManager.getCurrentUserId();
+        String postId = paramHelper.valueOf("postId");
+        if((bodyText.isEmpty() || bodyText == null) && (authorId.isEmpty() || authorId == null) && (postId.isEmpty() || postId == null)){
+            return "bad";
+        }
+        commentModel = new CommentModel();
+        commentModel.setPostId(postId);
+        commentModel.setAuthorId(authorId);
+        commentModel.setBodyText(bodyText);
+        commentModel.setDate();
+        mongoCollection.save(commentModel);
+        return postId;
+    });
+
+    public static Route getComments = ((request, response) -> {
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String postId = paramHelper.valueOf("postId");
+        String startAfterId = paramHelper.valueOf("startAfterId");
+        String date = paramHelper.valueOf("date");
+        PostModel post = mongoCollectionPost.findOne(eq("_id", new ObjectId(postId)));
+        List<CommentModel> commentModelList;
+        if(post == null){
+            return "post DNE";
+        }
+        if(startAfterId == null){
+            if(date == null){
+                Date thedate = new java.util.Date(System.currentTimeMillis());
+                DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+                date = dateFormat.format(thedate);
+            }
+            commentModelList = getByDate(postId, date);
```

Comment:
this will cause comments from the current day to be fetched. the spec requires that if `startAfterId` is missing, to return an initial list of ALL comments for the post, not just for the specific day